File: placesearch/src/main/java/org/tat/sdksample/model/SearchResults.java
package org.tat.sdksample.model;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.List;

public class SearchResults implements Parcelable {

    public List<SearchResult> searchResults;

    public SearchResults(List<SearchResult> searchResults){
        this.searchResults = searchResults;
    }


    protected SearchResults(Parcel in) {
        searchResults = in.createTypedArrayList(SearchResult.CREATOR);
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeTypedList(searchResults);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<SearchResults> CREATOR = new Creator<SearchResults>() {
        @Override
        public SearchResults createFromParcel(Parcel in) {
            return new SearchResults(in);
        }

        @Override
        public SearchResults[] newArray(int size) {
            return new SearchResults[size];
        }
    };
}

File: app/src/main/java/org/tat/sdksample/placesearch/model/SearchResults.kt
package org.tat.sdksample.placesearch.model

import android.os.Parcel
import android.os.Parcelable

data class SearchResults(var searchResults: List<SearchResult>? = null) : Parcelable {

    constructor(parcel: Parcel) : this(parcel.createTypedArrayList(SearchResult)) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeTypedList(searchResults)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<SearchResults> {
        override fun createFromParcel(parcel: Parcel): SearchResults {
            return SearchResults(parcel)
        }

        override fun newArray(size: Int): Array<SearchResults?> {
            return arrayOfNulls(size)
        }
    }


}
